wpiutil/src/main/java/edu/wpi/first/wpiutil/math/Nat.java

```diff
@@ -0,0 +1,130 @@
+/*----------------------------------------------------------------------------*/
+/* Copyright (c) 2019 FIRST. All Rights Reserved.                             */
+/* Open Source Software - may be modified and shared by FRC teams. The code   */
+/* must be accompanied by the FIRST BSD license file in the root directory of */
+/* the project.                                                               */
+/*----------------------------------------------------------------------------*/
+
+package edu.wpi.first.wpiutil.math;
+
+import edu.wpi.first.wpiutil.math.numbers.N0;
+import edu.wpi.first.wpiutil.math.numbers.N1;
+import edu.wpi.first.wpiutil.math.numbers.N10;
+import edu.wpi.first.wpiutil.math.numbers.N11;
+import edu.wpi.first.wpiutil.math.numbers.N12;
+import edu.wpi.first.wpiutil.math.numbers.N13;
+import edu.wpi.first.wpiutil.math.numbers.N14;
+import edu.wpi.first.wpiutil.math.numbers.N15;
+import edu.wpi.first.wpiutil.math.numbers.N16;
+import edu.wpi.first.wpiutil.math.numbers.N17;
+import edu.wpi.first.wpiutil.math.numbers.N18;
+import edu.wpi.first.wpiutil.math.numbers.N19;
+import edu.wpi.first.wpiutil.math.numbers.N2;
+import edu.wpi.first.wpiutil.math.numbers.N20;
+import edu.wpi.first.wpiutil.math.numbers.N3;
+import edu.wpi.first.wpiutil.math.numbers.N4;
+import edu.wpi.first.wpiutil.math.numbers.N5;
+import edu.wpi.first.wpiutil.math.numbers.N6;
+import edu.wpi.first.wpiutil.math.numbers.N7;
+import edu.wpi.first.wpiutil.math.numbers.N8;
+import edu.wpi.first.wpiutil.math.numbers.N9;
+
+/**
+ * A natural number expressed as a java class.
+ * The counterpart to {@link Num} that should be used as a concrete value.
+ *
+ * @param <T> The {@link Num} this represents.
+ */
+@SuppressWarnings({"MethodName", "unused", "PMD.TooManyMethods"})
+public interface Nat<T extends Num> {
+  /**
+   * The number this interface represents.
+   *
+   * @return The number backing this value.
+   */
+  int getNum();
+
+  static Nat<N0> N0() {
+    return N0.instance;
+  }
```

Comment:
After discussion, it would be great to see this file get auto generated with all of the imports and static methods